### ISSUE TITLE
Unable to add animation to implicit scrolling using accessibility focus.

### ISSUE THREAD

**Issue by FrobtheBuilder** [flutter/flutter#50303]
I'm currently adapting a Flutter app to work smoothly with Android TV devices. Documentation on this is sparse, but I've managed to feel my way through the process by using the accessibility focus system for directional navigation and control.

However, in one of my layouts, I'm using a `PageView` containing some grids of items, the grid separated out into pages for better directional navigation ergonomics. As expected, creating a `PageView` with `allowImplicitScrolling: true` allows you to navigate between pages by moving the selection reticle off the side of the grid. However, implicit scrolling in any context never has any animation accompanying it, just transitioning to the next view instantly. This ends up not looking very good.

```dart
Widget _buildPaginatedGrid(BuildContext build) {
    return SliverToBoxAdapter(
      child: Container(
        height: MediaQuery.of(build).size.height - gameHeroHeight(context),
        child: PageView(
          allowImplicitScrolling: true,
          controller: _pageController,
          children: [
            _buildRawGrid(build), //just display the same grid twice for testing
            _buildRawGrid(build)
          ]
        ),
      ),
    );
  }
```

I know this sort of scrolling is driven by calls to `RenderObject.showOnScreen`, obviously with a duration of zero, so I attempted to create a `SingleChildRenderObjectWidget` with a subclass of `RenderProxyBox` that overrides the `RenderObject.showOnScreen` method to force a duration. Yet, no matter where I put this thing, around each individual grid item, around the grid, around the `PageView`, no animation ever shows during implicit scrolling, and in fact, my override method is never even called. It's like the call never propagates to it no matter where in the widget hierarchy it's placed.

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

class RenderFocusSmoothScroll extends RenderProxyBox {
  RenderFocusSmoothScroll({RenderBox child}) : super(child);
  @override
  void showOnScreen({
    RenderObject descendant,
    Rect rect,
    Duration duration = Duration.zero,
    Curve curve = Curves.ease,
  }) {
    print("showOnScreen");
    super.showOnScreen(descendant: descendant, rect: rect, duration: Duration(milliseconds: 200), curve: Curves.ease);
  }
}

class FocusSmoothScroll extends SingleChildRenderObjectWidget {
  const FocusSmoothScroll({
    Key key,
    Widget child,
  })  : super(key: key, child: child);

  @override
  RenderFocusSmoothScroll createRenderObject(BuildContext context) {
    return RenderFocusSmoothScroll();
  }

  @override
  void updateRenderObject(BuildContext context, RenderFocusSmoothScroll renderObject) {
    
  }
}
```
Documentation on implicit scrolling using the focus system is a bit lacking, as is documentation on properly overriding `RenderObject` methods to enforce custom handling of these propagating calls, so some improvement here would be much appreciated. Ordinarily I would just ask about this on StackOverflow but I doubt many users there have dealt with this particular sort of issue, so I figured I would go straight to the Flutter team.

**Comment by iapicca** [comment]
Hi @FrobtheBuilder 

if you need assistance on personal code, 
please see https://flutter.dev/community for resources and asking questions like this,
you may also get some help if you post it on Stack Overflow.

If you believe the issue is within flutter itself
please provide your `flutter doctor -v` ,
and a runnable minimal code sample.
Thank you

**Comment by TahaTesser** [comment]
Hi @FrobtheBuilder 

Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.
